plugins/inputs/prometheus/prometheus.go

```diff
@@ -360,9 +367,71 @@ func (p *Prometheus) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
-func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error {
+func setResult(resultString string, requestFields map[string]interface{}, tags map[string]string) {
+	resultCodes := map[string]int{
+		"success":                 0,
+		"unable_to_decode":        1,
+		"body_read_error":         2,
+		"connection_failed":       3,
+		"timeout":                 4,
+		"dns_error":               5,
+		"http_code_not_ok":        6,
+		"error_reading_body":      7,
+		"content_length_exceeded": 8,
+	}
+
+	tags["result"] = resultString
+	requestFields["result_code"] = resultCodes[resultString]
+}
+
+func setError(err error, requestFields map[string]interface{}, tags map[string]string) error {
+	var timeoutError net.Error
+	if errors.As(err, &timeoutError) && timeoutError.Timeout() {
+		setResult("timeout", requestFields, tags)
+		return timeoutError
+	}
+
+	var urlErr *url.Error
+	if !errors.As(err, &urlErr) {
+		return nil
+	}
+
+	var opErr *net.OpError
+	if errors.As(urlErr, &opErr) {
+		var dnsErr *net.DNSError
+		var parseErr *net.ParseError
+
+		if errors.As(opErr, &dnsErr) {
+			setResult("dns_error", requestFields, tags)
+			return dnsErr
+		} else if errors.As(opErr, &parseErr) {
+			// Parse error has to do with parsing of IP addresses, so we
+			// group it with address errors
+			setResult("address_error", requestFields, tags)
+			return parseErr
+		}
+	}
+
+	return nil
+}
+
+func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) (map[string]interface{}, map[string]string, error) {
 	var req *http.Request
 	var uClient *http.Client
+	var metrics []telegraf.Metric
+	requestFields := make(map[string]interface{})
+	tags := map[string]string{}
+	u.OriginalURL.User = nil
```

Comment:
What is the purpose of this?

Reply:
@tguenneguez thanks for the updates - just another ping on this question.

Reply:
thanks for the updates - just another ping on this question.

Reply:
> thanks for the updates - just another ping on this question.

I think you meant to respond not just copy my response? :)

Reply:
Sorry, for what line exactly have you a question ?

Reply:
`u.OriginalURL.User = nil`

Reply:
Ok, I was searching on line :
> var metrics []telegraf.Metric
You have a keen eye ;-)
New commit